internal/blockchain/v0/reactor.go

```diff
@@ -85,10 +85,15 @@ type Reactor struct {
 	consReactor consensusReactor
 	fastSync    bool
 
-	blockchainCh  *p2p.Channel
-	peerUpdates   *p2p.PeerUpdates
-	peerUpdatesCh chan p2p.Envelope
-	closeCh       chan struct{}
+	blockchainCh *p2p.Channel
+	// blockchainOutBridgeCh is the bridge channel that other goroutine sending an Envelope to.
+	// processBlockchainCh will drain the Envelope from this channel and actually send it
+	// to blockchainCh.Out channel. This is necessary to prevent data race if other goroutine
+	// send Envelope directly to blockchainCh.Out channel, since when processBlockchainCh may
+	// close the blockchainCh.Out at the time other goroutine send data to channel.
```

Comment:
```suggestion
	// closed, the blockchainCh.Out at the time other goroutine send data to the channel.
```

Reply:
Hmm, I don't understand this suggestion. What I mean is that the processBlockchainCh function may close the blockchainCh.Out channel at the same time other goroutines send data to it.

Reply:
Yeah so the main point I want to get across is that the structure and grammar of this godoc for this field is a bit broken and hard to follow. Being that this change is critical to understand to the reader, it's imparitive that it's easy to understand as possible.

I suggest something like the following:

```go
// blockchainOutBridgeCh defines a channel that acts as a bridge between sending Envelope
// messages that the reactor will consume in processBlockchainCh and receiving messages
// from the peer updates channel and other goroutines. We do this instead of directly
// sending on blockchainCh.Out to avoid race conditions in the case where other goroutines
// send Envelopes directly to the to blockchainCh.Out channel, since processBlockchainCh
// may close the blockchainCh.Out channel at the same time that other goroutines send to
// blockchainCh.Out.
```